editor/editor_node.cpp

```diff
@@ -4940,24 +4984,31 @@ void EditorNode::_load_docks_from_config(Ref<ConfigFile> p_layout, const String
 				atidx = k;
 				break;
 			}
+
 			if (atidx == -1) { // Well, it's not anywhere.
```

Comment:
The problem of dock not getting unfloated on layout change comes from here. This code only looks for docks in dock slots, but floating docks are elsewhere. You need to add a search in `floating_docks` too.

Reply:
Now I get what you have tested and why it doesn't work. I didn't think about restoring layout with docks currently floating.
I'll look to it.